src/ipolatev.f90

```diff
@@ -409,163 +402,130 @@ SUBROUTINE IPOLATEV_grib2(IP,IPOPT,IGDTNUMI,IGDTMPLI,IGDTLENI, &
+  !! @param[in] mo    Skip number between output grid fields if km>1 or dimension of output grid fields if km=1.
+  !! @param[in] km    Number of fields to interpolate.
+  !! @param[in] ibi   Input bitmap flags.
+  !! @param[in] li    Input bitmaps (if respective ibi(k)=1).
+  !! @param[in] ui    Input u-component fields to interpolate.
+  !! @param[in] vi    Input v-component fields to interpolate.
+  !! @param[out] no Number of output points (only if kgdso(1)<0).
+  !! @param[out] rlat Output latitudes in degrees (if kgdso(1)<0).
+  !! @param[out] rlon Output longitudes in degrees (if kgdso(1)<0).
+  !! @param[inout] crot Vector rotation cosines (if igdtnumo>=0).
+  !! @param[inout] srot Vector rotation sines (if igdtnumo>=0).
+  !! @param[out] ibo Output bitmap flags.
+  !! @param[out] lo  Output bitmaps (always output).
+  !! @param[out] uo  Output u-component fields interpolated.
+  !! @param[out] vo  Output v-component fields interpolated.
+  !! @param[out] iret Return code.
+  !! - 0 Successful interpolation.
+  !! - 1 Unrecognized interpolation method.
+  !! - 2 Unrecognized input grid or no grid overlap.
+  !! - 3 Unrecognized output grid.
+  !! - 1x Invalid bicubic method parameters.
+  !! - 3x Invalid budget method parameters.
+  !! - 4x Invalid spectral method parameters.
+  !!
+  !! @note Examples demonstrating relative cpu costs.
+  !! This example is interpolating 12 levels of winds
+  !! from the 360 x 181 global grid (ncep grid 3)
+  !! to the 93 x 68 hawaiian mercator grid (ncep grid 204).
+  !!
+  !! The example times are for the c90. As a reference, the cp time
+  !! for unpacking the global 12 temperature fields is 0.07 seconds.
+  !!
+  !!   METHOD  | IP| IPOPT       |  CP SECONDS
+  !!   --------| --|-------------| ----------
+  !!   BILINEAR| 0 |             |   0.05
+  !!   BICUBIC | 1 | 0           |   0.16
+  !!   BICUBIC | 1 | 1           |   0.17
+  !!   NEIGHBOR| 2 |             |   0.02
+  !!   BUDGET  | 3 | -1,-1       |   0.94
+  !!   SPECTRAL| 4 | 0,40        |   0.31
+  !!   SPECTRAL| 4 | 1,40        |   0.33
+  !!   SPECTRAL| 4 | 0,-1        |   0.59
+  !!   N-BUDGET| 6 | -1,-1       |   0.31
+  !!
+  !!   The spectral interpolation is fast for the mercator grid.
+  !!   However, for some grids the spectral interpolation is slow.
+  !!
+  !!   The following example is interpolating 12 levels of temperatures
+  !!   from the 360 x 181 global grid (ncep grid 3)
+  !!   to the 93 x 65 conus lambert conformal grid (ncep grid 211).
+  !!
+  !!   METHOD  | IP| IPOPT        |CP SECONDS
+  !!   --------| --| -------------|----------
+  !!   BILINEAR| 0 |              | 0.05
+  !!   BICUBIC | 1 | 0            | 0.15
+  !!   BICUBIC | 1 | 1            | 0.16
+  !!   NEIGHBOR| 2 |              | 0.02
+  !!   BUDGET  | 3 | -1,-1        | 0.92
+  !!   SPECTRAL| 4 | 0,40         | 4.51
+  !!   SPECTRAL| 4 | 1,40         | 5.77
+  !!   SPECTRAL| 4 | 0,-1         | 12.60
+  !!   N-BUDGET| 6 | -1,-1        | 0.33
```

Comment:
Not a comment on your doxygenation, but what scares me is testing this kind of function. I rather suspect that many of these options are never used anywhere, but all will have to be tested.